content/app/models/file-path.js

```diff
@@ -0,0 +1,90 @@
+define("content/app/models/file-path", function(require, exports, module) {
+
+	/**
+	 * Simple class used for handling of file paths.
+	 */
+	var FilePath = Class({
+
+		$statics : {
+			FOLDER_SEPARATOR : '/',
+			EXTENSION_SEPARATOR : '.',
+		},
+
+		/**
+		 * Class constructor
+		 * @param  	{String}	path 	Full file path.
+		 */
+		constructor : function(fullPath) {
+			this._fullPath = fullPath;
```

Comment:
You could do filepath parsing in constructor or in some method that is called from constructor.
You don't need to parse it on every get, and beside that getters are doing the same things...
You could do here:
1. split by path separator and you got pathSegments.
2. last segment is filename
3. split filename and you can set extension and short filename (or how it should be called)

That should be few lines that sets all local props.
If you want to give proper names for properties take a look at some other libs:
https://docs.python.org/2/library/os.path.html
http://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html
No need to camel-casing word filename. FileExtension can just be extension. Fullpath can only be path, etc....